rr/gstreamer/gst_media.py

```diff
@@ -104,6 +130,31 @@ def stop_media(self):
         if gst.StateChangeReturn.FAILURE == ret:
             raise GstMediaError("Unable to stop the media")
 
+    def install_callback(self, callback):
+        if callback is None:
+            raise GstMediaError("Invalid callback")
+
+        self.callback = callback
+
+    def on_new_buffer(self):
+        try:
+            appsink = self._pipeline.get_by_name("appsink")
+            appsink.connect("new-sample", self._on_new_buffer, appsink)
+            time.sleep(1)
+        except AttributeError as e:
+            raise GstMediaError("Unable to get sample from buffer") from e
+
+    def _on_new_buffer(self, appsink, data):
+        sample = appsink.emit("pull-sample")
+
+        image_array = construct_image_from_gst_sample(sample)
+        if image_array is None:
+            return gst.FlowReturn.ERROR
```

Comment:
We are reporting errors as exceptions.

Reply:
In this case, this return code it's for the AppSink connect, and it actually expects it to be GstFlowReturn type.